swagger.js

```diff
@@ -31,6 +31,34 @@ const doc = {
         },
       },
     },
+    UserWithAlbums: {
+      
+        $name: 'John Doe',
+        $username: 'johndoe',
+        $email: 'john@example.com',
+        $role: { '@enum': ['user', 'admin'] },
+        profileImage: {
+          url: 'https://example.com/profile-image.jpg',
+        creditCardInfo: {
+          hashedNumber: '1234567890',
+          expiry: '12/24',
+          preferredPaymentOption: {
+            '@enum': ['credit card', 'paypal', 'google pay'],
+          },
+        },
+        },
+        
+        $artistName: 'Britney Spears',
+        $albumName: '...Baby one more time',
+        price: 20,
+        image: '/uploads/example.jpg',
+        releaseDate: '2023-07-28 T00:00:00.000+00:00',
+        category: 'pop',
+        spotifyUrl: 'https://api.spotify.com/v1/albums/6r1lh7fHMB499vGKtIyJLy',
+        averageRating: 4,
+        numOfReviews: 2,
+      
+    },
```

Comment:
I think some of your brackets got a bit mis-aligned here. 

Looking at a request I made on this endpoint in the docs (you could alternatively do the same in Postman)

![](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/a60086f8-7d97-405a-b322-54b77c12d712)

the shape of the response is:

```
{
  "user": {
    "_id": "64d6ca92a15d2e18ab96a2a3",
    "name": "Akos123123",
    "email": "akos123@example.com",
    "role": "user",
    "__v": 0,
    "username": "akos92",
    "purchasedAlbums": [
      {
        "_id": "64ef50c8c5551074444547bc",
        "album": {
          "_id": "64d2a94c793389a43fc5a8d6",
          "artistName": "F",
          "albumName": "AUSTIN",
          "price": 0,
          "image": "https://i.scdn.co/image/ab67616d0000b27371cae34ad5a39bdab78af13e",
          "releaseDate": "2023-07-28T00:00:00.000Z",
          "spotifyUrl": "https://api.spotify.com/v1/albums/6r1lh7fHMB499vGKtIyJLy",
          "averageRating": 0,
          "numOfReviews": 0,
          "__v": 0,
          "createdAt": "2023-08-08T20:45:00.942Z",
          "updatedAt": "2023-08-17T10:10:42.861Z",
          "id": "64d2a94c793389a43fc5a8d6"
        },
        "user": "64d6ca92a15d2e18ab96a2a3",
        "__v": 0
      }
    ],
    "id": "64d6ca92a15d2e18ab96a2a3"
  },
  "purchasedAlbumCount": 1
}
```

We can basically copy that into this swagger.js file:

```suggestion
    UserWithAlbums: {
        $id: '64d6ca92a15d2e18ab96a2a3',
        $name: 'John Doe',
        $username: 'johndoe',
        $email: 'john@example.com',
        $role: { '@enum': ['user', 'admin'] },
        profileImage: {
            url: 'https://example.com/profile-image.jpg',
        },
        creditCardInfo: {
            hashedNumber: '1234567890',
            expiry: '12/24',
            preferredPaymentOption: {
                '@enum': ['credit card', 'paypal', 'google pay'],
            },
        },
        $purchasedAlbums: [{
            $_id: '64ef50c8c5551074444547bc',
            $user: '64d6ca92a15d2e18ab96a2a3',
            $album: {
                $id: '64d2a94c793389a43fc5a8d6',
                $albumName: '...Baby one more time',
                $artistName: 'Britney Spears',
                $createdAt: '2023-08-08T20:45:00.942Z',
                $updatedAt: '2023-08-17T10:10:42.861Z',
                $price: 20.99,
                image: 'https://i.scdn.co/image/ab67616d0000b27371cae34ad5a39bdab78af13e',
                spotifyUrl: 'https://api.spotify.com/v1/albums/6r1lh7fHMB499vGKtIyJLy',
                releaseDate: '2023-07-28T00:00:00.000Z',
                averageRating: 0,
                numOfReviews: 0,
            },
        }],
    },
```

Then, in the controller, we can tell swagger-autogen to use this as the response for the `withAlbums` endpoint by saying: 

```
  /*
     #swagger.summary = 'Fetch a user by id and all albums they purchased.'
     #swagger.responses[200] = {
				description: 'User and their albums purchased fetched successfully.',
				schema: { user: { $ref: '#/definitions/UserWithAlbums' }, purchasedAlbumCount: 1}
		 }
		  #swagger.responses[404] = { description: 'No user with id found.' }
  */
```

Reply:
@akosasante thank you, Akos, just a little question- to test this rote I login, create an order but when a mak
![Captura de pantalla 2023-08-31 a las 11 42 50](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/103755090/1c80bfae-cc1a-49ba-9c61-59244f6ea59d)
![Uploading Captura de pantalla 2023-08-31 a las 11.42.12.png…]()
e a GET request. to http://localhost:8000/api/v1/users/showMe/withAlbums I don't get album that you do in swagger (on the image)- what am I doing wrong?

Reply:
This is expected. We aren't yet creating the `PurchasedAlbum` object when creating the order -- see this ticket: https://codethedreamteam4-2023.atlassian.net/browse/PS-125

For testing I just created a PurchasedAlbum object for my user in a node console. You could also do it directly in the Mongo Atlas UI

Reply:
@akosasante wow!!! thank you-i've just tried it (creating an entry in mongo db). just curious- to create a PurchasedAlbum object from node console did you do smth like -in a root  added newFileForCreatingPurchasedAlbum.js  and   executed it it with `node  newFileForCreatingPurchasedAlbum.js` - ```
const mongoose = require('mongoose');
require('dotenv').config();
const PurchasedAlbum = require('./src/models/PurchasedAlbum');

// Connect to the MongoDB database
mongoose.connect(process.env.MONGO_URL, { useNewUrlParser: true, useUnifiedTopology: true })
  .then(() => {
    console.log('Connected to the database');
    
    // Replace with the actual user and album IDs
    const userId = '111111111';
    const albumId = '2222222';
    
    // Create a PurchasedAlbum object
    const purchasedAlbum = new PurchasedAlbum({
      album: albumId,
      user: userId
    });

    // Save the purchased album to the database
    purchasedAlbum.save()
      .then(savedPurchasedAlbum => {
        console.log('PurchasedAlbum saved:', savedPurchasedAlbum);
        mongoose.disconnect();
      })
      .catch(error => {
        console.error('Error saving PurchasedAlbum:', error);
        mongoose.disconnect();
      });
  })
  .catch(error => {
    console.error('Error connecting to the database:', error);
  });
```
?